H/HDF5/build_tarballs.jl

```diff
@@ -1,90 +1,400 @@
-platforms = [
-    Platform("x86_64", "linux"),
-    Platform("aarch64", "linux"; libc="glibc"),
-    Platform("x86_64", "macos"),
-    Platform("x86_64", "windows"),
-    Platform("i686", "windows"),
-    Platform("aarch64", "macos"),
-]
+platforms = supported_platforms()
+
+# Platforms:
+# aarch64-apple-darwin
+# aarch64-linux-gnu
+# aarch64-linux-musl
+# armv6l-linux-gnueabihf
+# armv6l-linux-musleabihf
+# armv7l-linux-gnueabihf
+# armv7l-linux-musleabihf
+# i686-linux-gnu
+# i686-linux-musl
+# i686-w64-mingw32
+# powerpc64le-linux-gnu
+# x86_64-apple-darwin
+# x86_64-linux-gnu
+# x86_64-linux-musl
+# x86_64-unknown-freebsd
+# x86_64-w64-mingw32
 
 # The products that we will ensure are always built
 products = [
+    # HDF5 tools
+    ExecutableProduct("h5c++", :h5cxx),
+    ExecutableProduct("h5cc", :h5cc),
+    ExecutableProduct("h5clear", :h5clear),
+    ExecutableProduct("h5copy", :h5copy),
+    ExecutableProduct("h5debug", :h5debug),
+    ExecutableProduct("h5delete", :h5delete),
+    ExecutableProduct("h5diff", :h5diff),
+    ExecutableProduct("h5dump", :h5dump),
+    ExecutableProduct("h5fc", :h5fc),
+    ExecutableProduct("h5format_convert", :h5format_convert),
+    ExecutableProduct("h5import", :h5import),
+    ExecutableProduct("h5jam",:h5jam),
+    ExecutableProduct("h5ls", :h5ls),
+    ExecutableProduct("h5mkgrp", :h5mkgrp),
+    ExecutableProduct("h5perf_serial",:h5perf_serial),
+    ExecutableProduct("h5redeploy", :h5redeploy),
+    ExecutableProduct("h5repack", :h5repack),
+    ExecutableProduct("h5repart", :h5repart),
+    ExecutableProduct("h5stat", :h5stat),
+    ExecutableProduct("h5unjam", :h5unjam),
+    ExecutableProduct("h5watch", :h5watch),
+
+    # HDF5 libraries
     LibraryProduct("libhdf5", :libhdf5),
+    LibraryProduct("libhdf5_cpp", :libhdf5_cpp),
+    LibraryProduct("libhdf5_fortran", :libhdf5_fortran),
     LibraryProduct("libhdf5_hl", :libhdf5_hl),
+    LibraryProduct("libhdf5_hl_cpp", :libhdf5_hl_cpp),
+    LibraryProduct("libhdf5hl_fortran", :libhdf5_hl_fortran),
 ]
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    Dependency("Zlib_jll"),
-    Dependency("OpenSSL_jll"; compat="1.1.10"),
+    # For OpenMP we use libomp from `LLVMOpenMP_jll` where we use LLVM as compiler (BSD 
+    # systems), and libgomp from `CompilerSupportLibraries_jll` everywhere else. 
+    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae");
+               platforms=filter(!Sys.isbsd, platforms)),
+    Dependency(PackageSpec(name="LLVMOpenMP_jll", uuid="1d63c593-3942-5779-bab2-d838dc0a180e");
+               platforms=filter(Sys.isbsd, platforms)),
+    # LibCURL_jll version 8 does not work on *-musl architectures. The respective shared library can't be loaded.
+    # LibCURL_jll version 7 does not exist for armv6l-linux-musleabihf.
+    # Dependency("LibCURL_jll"; compat="7.66.0"),
     Dependency("LibCURL_jll"),
```

Comment:
This is probably better:
```suggestion
    Dependency("LibCURL_jll"; compat="7.73.0, 8"),
```

Reply:
Not sure. LibCURL 8 exists for `hdf5-x86-64-linux-musl`, but the library can't be opened (`ldd` complains). I'll try.